app.unregister() fails for instance method callbacks

In [app.py](https://github.com/tryexceptpass/sofi/blob/master/sofi/app/app.py#L101), the key used for callbacks relies on the `id()` function of the callback.  The internal id for instance methods defined with `async def` apparently does not seem to be persistent.
```python
import asyncio

class A(object):
    async def foo(self):
        print(id(self.bar))
    async def bar(self):
        pass    

loop = asyncio.get_event_loop()

async def test():
    a = A()
    await a.foo()
    await a.foo()

loop.run_until_complete(test())

# 140002553590152
# 140002529194440
```
Would it be a reasonable solution to check callbacks during `register()` for `isinstance(callback, types.MethodType)` and use a combination of `id(callback.__func__)` and `id(callback.__self__)` as a key?

Update...
So for some reason now I cannot seem to duplicate the results from the snippet above.  The issue is still present in my application though (the first line is the output from `print(id(self.<mycallback>))`:
```
140280319751880
2017-01-06 16:29:39,069 [ERROR] - default_exception_handler: Task exception was never retrieved
future: <Task finished coro=<SofiEventProtocol.onMessage() done, defined at /home/nocarrier/programs/vidhub-control/venv3/lib/python3.5/site-packages/sofi/app/app.py:162> exception=KeyError('140280319750728',)>
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/home/nocarrier/programs/vidhub-control/venv3/lib/python3.5/site-packages/sofi/app/app.py", line 170, in onMessage
    await self.processor.process(self, body)
  File "/home/nocarrier/programs/vidhub-control/venv3/lib/python3.5/site-packages/sofi/app/app.py", line 145, in process
    await handler(event)
  File "vidhubcontrol/sofi_ui/main.py", line 335, in on_device_select_click
    self.app.unregister('click', self.on_click, selector='button')
  File "/home/nocarrier/programs/vidhub-control/venv3/lib/python3.5/site-packages/sofi/app/app.py", line 29, in unregister
    self.interface.unregister(event, callback, selector)
  File "/home/nocarrier/programs/vidhub-control/venv3/lib/python3.5/site-packages/sofi/app/app.py", line 125, in unregister
    self.handlers[event].pop(str(id(callback)))
KeyError: '140280319750728'
```

--- a/sofi/app/app.py
+++ b/sofi/app/app.py
@@ -122,7 +122,7 @@ class SofiEventProcessor(object):
         if selector is None:
             self.handlers[event]['_'].remove(callback)
         else:
-            self.handlers[event].pop(str(id(callback)))
+            self.handlers[event].get(str(id(callback)), []).remove(callback)
 
         if event not in ('init', 'load', 'close'):
             self.dispatch({ 'name': 'unsubscribe', 'event': event, 'selector': selector, 'key': str(id(callback)) })

--- a/sofi/app/app.py
+++ b/sofi/app/app.py
@@ -122,7 +122,9 @@ class SofiEventProcessor(object):
         if selector is None:
             self.handlers[event]['_'].remove(callback)
         else:
-            self.handlers[event].get(str(id(callback)), []).remove(callback)
+            handler_list = self.handlers[event].get(str(id(callback)), [])
+            if callback in handler_list:
+                handler_list.remove(callback)
 
         if event not in ('init', 'load', 'close'):
             self.dispatch({ 'name': 'unsubscribe', 'event': event, 'selector': selector, 'key': str(id(callback)) })

